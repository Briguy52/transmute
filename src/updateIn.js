import curry from "./curry";

/**
 * Apply `updater` to the value at `keyPath`.
 *
 * @param  {Array<any>} keyPath the location where `updater` should be applied.
 * @param  {Function} updater the tranformation to apply.
 * @param  {Iterable} subject the thing to update.
 * @return {Iterable}
 */
function updateIn(keyPath, updater, subject) {
  return subject.updateIn(keyPath, updater);
}

export default curry(updateIn);
