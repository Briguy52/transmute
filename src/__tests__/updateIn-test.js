import { Map } from "immutable";
import updateIn from "../updateIn";

describe("transmute/updateIn", () => {
  it("updates a nested Map", () => {
    const incTwo = updateIn(["one", "two"], n => n + 1);
    expect(
      incTwo(
        Map({
          one: Map({
            two: 2
          })
        })
      )
    ).toMatchSnapshot();
  });
});
