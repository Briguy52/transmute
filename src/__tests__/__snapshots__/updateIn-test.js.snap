// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/updateIn updates a nested Map 1`] = `
Immutable.Map {
  "one": Immutable.Map {
    "two": 3,
  },
}
`;
